src/services/services.ts

```diff
@@ -4584,7 +4584,7 @@ namespace ts {
 
                 forEach(signatureDeclarations, d => {
                     if ((selectConstructors && d.kind === SyntaxKind.Constructor) ||
-                        (!selectConstructors && (d.kind === SyntaxKind.FunctionDeclaration || d.kind === SyntaxKind.MethodDeclaration || d.kind === SyntaxKind.MethodSignature))) {
+                        (!selectConstructors && (d.kind === SyntaxKind.FunctionDeclaration || d.kind === SyntaxKind.MethodDeclaration || d.kind === SyntaxKind.MethodSignature || d.kind === SyntaxKind.VariableDeclaration))) {

```

Comment:
i do not think this is the correct fix. this function tries to get you on the implementation signature or the last one if no implementation was found. this is special to signatures, this issue is about filtering the declaration location based on the reference location kind. i.e.:

```ts
var foo;   // value
type foo = any;  // type
namespace foo {} // namespace

+foo; // should only see the value (i.e var foo;)

var x: foo; // should only see the type (i.e type foo = any;)

foo.blah; // should see both namespace and value (i.e var foo, and namespace foo).
```

what we need to do is filter the declarations based on the reference node. 

we have two functions that do that already, 1. `getMeaningFromLocation` see https://github.com/Microsoft/TypeScript/blob/master/src/services/services.ts#L6497, this tells you which one of value, type, or namespace the reference node is, and 2. `getMeaningFromDeclaration` see https://github.com/Microsoft/TypeScript/blob/master/src/services/services.ts#L6368, this tells you what is the meaning of a declaration. 
the declaration location we add should be the ones the match:
`symbol.declarations.filter(d => getMeaningFromDeclaration(d) & getMeaningFromLocation(node) !== 0)`